### PR TITLE
community: Add support for metadata indexing policy in Cassandra vector store

### PR DESCRIPTION
**Pull request by hemidactylus**
This PR adds a constructor `metadata_indexing` parameter to the Cassandra vector store to allow optional fine-tuning of which fields of the metadata are to be indexed.

This is a feature supported by the underlying CassIO library. Indexing mode of "all", "none" or deny- and allow-list based choices are available.

The rationale is, in some cases it's advisable to programmatically exclude some portions of the metadata from the index if one knows in advance they won't ever be used at search-time. this keeps the index more lightweight and performant and avoids limitations on the length of _indexed_ strings.

I added a integration test of the feature. I also added the possibility of running the integration test with Cassandra on an arbitrary IP address (e.g. Dockerized), via `CASSANDRA_CONTACT_POINTS=10.1.1.5,10.1.1.6 poetry run pytest [...]` or similar.

While I was at it, I added a line to the `.gitignore` since the mypy _test_ cache was not ignored yet.

My X (Twitter) handle: @rsprrs.

### CODE REVIEW

**Comment by hemidactylus** [comment]
@cbornet FYI ^

File: libs/community/langchain_community/vectorstores/cassandra.py [hunk 1]
```diff
@@ -120,12 +132,15 @@ def __init__(
...
+        if metadata_indexing is None:
+            metadata_indexing = "all"
+
         self.table = MetadataVectorCassandraTable(
             session=session,
             keyspace=keyspace,
             table=table_name,
             vector_dimension=embedding_dimension,
-            metadata_indexing="all",
+            metadata_indexing=metadata_indexing,
```
Comment (cbornet) [inline libs/community/langchain_community/vectorstores/cassandra.py:143]:
```suggestion
            metadata_indexing=metadata_indexing or "all",
```

File: libs/community/langchain_community/vectorstores/cassandra.py [hunk 2]
```diff
@@ -59,6 +59,7 @@ def __init__(
         *,
         body_index_options: Optional[List[Tuple[str, Any]]] = None,
         setup_mode: SetupMode = SetupMode.SYNC,
+        metadata_indexing: Optional[Union[Tuple[str, Iterable[str]], str]] = None,
```
Comment (cbornet) [inline libs/community/langchain_community/vectorstores/cassandra.py:62]:
```suggestion
        metadata_indexing: Optional[Union[Tuple[str, Iterable[str]], str]] = "all",
```

File: libs/community/langchain_community/vectorstores/cassandra.py [hunk 3]
```diff
@@ -885,6 +900,7 @@ def from_texts(
         batch_size: int = 16,
         ttl_seconds: Optional[int] = None,
         body_index_options: Optional[List[Tuple[str, Any]]] = None,
+        metadata_indexing: Optional[Union[Tuple[str, Iterable[str]], str]] = None,
```
Comment (cbornet) [inline libs/community/langchain_community/vectorstores/cassandra.py:903]:
```suggestion
        metadata_indexing: Union[Tuple[str, Iterable[str]], str] = "all",
```

File: libs/community/langchain_community/vectorstores/cassandra.py [hunk 4]
```diff
@@ -935,6 +952,7 @@ async def afrom_texts(
         concurrency: int = 16,
         ttl_seconds: Optional[int] = None,
         body_index_options: Optional[List[Tuple[str, Any]]] = None,
+        metadata_indexing: Optional[Union[Tuple[str, Iterable[str]], str]] = None,
```
Comment (cbornet) [inline libs/community/langchain_community/vectorstores/cassandra.py:955]:
```suggestion
        metadata_indexing: Union[Tuple[str, Iterable[str]], str] = "all",
```

File: libs/community/langchain_community/vectorstores/cassandra.py [hunk 5]
```diff
@@ -985,6 +1004,7 @@ def from_documents(
         batch_size: int = 16,
         ttl_seconds: Optional[int] = None,
         body_index_options: Optional[List[Tuple[str, Any]]] = None,
+        metadata_indexing: Optional[Union[Tuple[str, Iterable[str]], str]] = None,
```
Comment (cbornet) [inline libs/community/langchain_community/vectorstores/cassandra.py:1007]:
```suggestion
        metadata_indexing: Union[Tuple[str, Iterable[str]], str] = "all",
```

File: libs/community/langchain_community/vectorstores/cassandra.py [hunk 6]
```diff
@@ -1036,6 +1057,7 @@ async def afrom_documents(
         concurrency: int = 16,
         ttl_seconds: Optional[int] = None,
         body_index_options: Optional[List[Tuple[str, Any]]] = None,
+        metadata_indexing: Optional[Union[Tuple[str, Iterable[str]], str]] = None,
```
Comment (cbornet) [inline libs/community/langchain_community/vectorstores/cassandra.py:1060]:
```suggestion
        metadata_indexing: Union[Tuple[str, Iterable[str]], str] = "all",
```

File: libs/community/tests/integration_tests/vectorstores/test_cassandra.py [hunk 7]
```diff
@@ -19,13 +21,24 @@ def _vectorstore_from_texts(
     metadatas: Optional[List[dict]] = None,
     embedding_class: Type[Embeddings] = ConsistentFakeEmbeddings,
     drop: bool = True,
+    metadata_indexing: Optional[Union[Tuple[str, Iterable[str]], str]] = None,
```
Comment (cbornet) [inline libs/community/tests/integration_tests/vectorstores/test_cassandra.py:24]:
```suggestion
    metadata_indexing: Union[Tuple[str, Iterable[str]], str] = "all",
```

File: libs/community/tests/integration_tests/vectorstores/test_cassandra.py [hunk 8]
```diff
@@ -53,13 +67,24 @@ async def _vectorstore_from_texts_async(
...
```
Comment (cbornet) [inline libs/community/tests/integration_tests/vectorstores/test_cassandra.py:70]:
```suggestion
    metadata_indexing: Union[Tuple[str, Iterable[str]], str] = "all",
```

File: libs/community/tests/integration_tests/vectorstores/test_cassandra.py [hunk 9]
```diff
@@ -19,13 +21,24 @@ def _vectorstore_from_texts(
...
+    table_name: Optional[str] = None,
```
Comment (cbornet) [inline libs/community/tests/integration_tests/vectorstores/test_cassandra.py:25]:
```suggestion
    table_name: str = ""vector_test_table"",
```

File: libs/community/tests/integration_tests/vectorstores/test_cassandra.py [hunk 10]
```diff
@@ -19,13 +21,24 @@ def _vectorstore_from_texts(
...
 ) -> Cassandra:
     from cassandra.cluster import Cluster
 
     keyspace = "vector_test_keyspace"
-    table_name = "vector_test_table"
+    if table_name is None:
+        table_name = "vector_test_table"
```
Comment (cbornet) [inline libs/community/tests/integration_tests/vectorstores/test_cassandra.py:31]:
```suggestion
```

File: libs/community/tests/integration_tests/vectorstores/test_cassandra.py [hunk 11]
```diff
@@ -53,13 +67,24 @@ async def _vectorstore_from_texts_async(
     metadatas: Optional[List[dict]] = None,
     embedding_class: Type[Embeddings] = ConsistentFakeEmbeddings,
     drop: bool = True,
+    metadata_indexing: Optional[Union[Tuple[str, Iterable[str]], str]] = None,
+    table_name: Optional[str] = None,
```
Comment (cbornet) [inline libs/community/tests/integration_tests/vectorstores/test_cassandra.py:71]:
```suggestion
    table_name: Optional[str] = "vector_test_table",
```

File: libs/community/tests/integration_tests/vectorstores/test_cassandra.py [hunk 12]
```diff
@@ -53,13 +67,24 @@ async def _vectorstore_from_texts_async(
...
 ) -> Cassandra:
     from cassandra.cluster import Cluster
 
     keyspace = "vector_test_keyspace"
-    table_name = "vector_test_table"
+    if table_name is None:
+        table_name = "vector_test_table"
```
Comment (cbornet) [inline libs/community/tests/integration_tests/vectorstores/test_cassandra.py:77]:
```suggestion
```

File: libs/community/langchain_community/vectorstores/cassandra.py [hunk 13]
```diff
@@ -125,7 +137,7 @@ def __init__(
             keyspace=keyspace,
             table=table_name,
             vector_dimension=embedding_dimension,
-            metadata_indexing="all",
+            metadata_indexing=metadata_indexing or "all",
```
Comment (cbornet) [inline libs/community/langchain_community/vectorstores/cassandra.py:140]:
```suggestion
            metadata_indexing=metadata_indexing,
```

**Review by cbornet — APPROVED** [review]

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**